Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift

```diff
@@ -589,6 +591,25 @@ extension ProjectDescription.Settings {
         if !swiftFlags.isEmpty {
             settingsDictionary["OTHER_SWIFT_FLAGS"] = .array(["$(inherited)"] + swiftFlags)
         }
+        if let cLanguageStandard = packageInfo.cLanguageStandard {
+            settingsDictionary["GCC_C_LANGUAGE_STANDARD"] = .string(cLanguageStandard)
+        }
+        if let cxxLanguageStandard = packageInfo.cxxLanguageStandard {
+            settingsDictionary["CLANG_CXX_LANGUAGE_STANDARD"] = .string(cxxLanguageStandard)
+        }
+        if let swiftLanguageVersion = packageInfo.swiftLanguageVersions?
+            .sorted(by: {
+                // If not a valid version, put it at the beginning of the sort
+                guard let lDouble = Double($0) else { return true }
+                guard let rDouble = Double($1) else { return false }
+
+                // If valid versions, sort ascending and take the latest
+                return lDouble < rDouble
+            })
+            .last
+        {
+            settingsDictionary["SWIFT_VERSION"] = .string(swiftLanguageVersion)
```

Comment:
Why does SPM support multiple versions? Am I right assuming that we take the oldest one from the list? It'd be great to add some context here around that, I don't think it's obvious by looking at the code why SPM does something like this.

Reply:
It takes the latest one, as long as is smaller than the configured one, adding that logic as well